### PR TITLE
Feat: implement temporary scoring using SEMMED NGD

### PR DESCRIPTION
**Pull request by ericz1803**
(Related to https://github.com/biothings/BioThings_Explorer_TRAPI/issues/368)

@andrewsu suggets that it would be useful to at least return some scores back to the user. This PR incorporates SEMMED NGD to calculate scores (more details in linked issue).

@colleenXu could you do some basic testing to make sure the top scored results make sense biologically? Also, the default value for score not found will be 0 with this change, as suggested by your comment.

### CODE REVIEW

**Comment by colleenXu** [comment]
I don't think this is ready yet.

A. when I try out this 1-hop query, there's an error getting scores. I've included console logs that look relevant below...

```
bte:biothings-explorer-trapi:QueryResult Error getting scores:  Error: Request failed with status code 400

    data: {
      code: 400,
      success: false,
      error: 'Bad Request',
      keyword: 'umls',
      max: 1000,
      size: 1394
    }
```

<details>
<summary>1-hop query: demo B.2f</summary>

```
{
    "message": {
        "query_graph": {
            "nodes": {
                "n0": {
                    "ids": ["CHEMBL.COMPOUND:CHEMBL50"],
                    "categories": ["biolink:ChemicalEntity"],
                    "name": "Quercetin"
                },
                "n1": {
                    "categories": ["biolink:DiseaseOrPhenotypicFeature"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "n0",
                    "object": "n1",
                    "predicates": ["biolink:treats", "biolink:has_real_world_evidence_of_association_with"]
                }
            }
        }
    }
}
```
</details>

B. [EDIT, still in progress of testing] not sure if this will work with a multi-hop query...

C. It would be good to have some TRAPI logs related to the scoring process. Right now the logs are only in the console.

**Comment by colleenXu** [comment]
Queries that I think are working:

<details>
<summary>1-hop query Explain: demo D.3</summary>

```
{
  "message": {
    "query_graph": {
      "edges": {
        "e00": {
          "subject": "n00",
          "object": "n01"
        }
      },
      "nodes": {
        "n00": {
          "ids": [
            "DRUGBANK:DB00215",
            "DRUGBANK:DB01175",
            "DRUGBANK:DB00472",
            "DRUGBANK:DB00176",
            "DRUGBANK:DB00715",
            "DRUGBANK:DB01104"
          ],
          "categories": ["biolink:SmallMolecule"]
        },
        "n01": {
          "ids": [
              "DOID:5844", 
              "DOID:1936", 
              "DOID:3393"
            ],
          "categories": ["biolink:Disease"]
        }
      }
    }
  }
}
```
</details>

**Comment by colleenXu** [comment]
I'll also note for @andrewsu that it looks like ARAX will replace a score of 0 with n/a. This is what situation B's results look like in ARAX.
Screen Shot 2022-03-23 at 4 34 18 PM

vs the current (score of 1 default):
Screen Shot 2022-03-23 at 4 37 26 PM

File: src/score.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const axios = require('axios');
+const GraphHelper = require('./helper');
+const helper = new GraphHelper();
+
+async function getScores (dataByEdge) {
+  let pairs = {};
+
+  let combosWithoutResults = 0;
+
+  Object.keys(dataByEdge).forEach((edge_key) => {
```
Comment (tokebe) [inline src/score.js:11]: `dataByEdge` could be `recordsByQEdgeID` to match vocab standard.

**Comment by ericz1803** [comment]
Good catch! The issue with querying more than 1000 combos is fixed. I tried the demo D.1 query on the normal `v1/query` endpoint and it does return scores. I think the reason there were no scores with the limited query is that the scores for each edge get multiplied together, so if there aren't scores for both edges for a particular result, it will also end up as 0.

I also fixed the variable name to match the standard.

**Comment by colleenXu** [comment]
I like the update.

I think it's an okay attempt at scoring. I can tell that it depends on the level of UMLS cross-references for ID namespaces, and the amount of coverage in semmeddb (some associations are kind of boring but ranked highly because they were annotated a lot).

**Pull request merged**